R/target_market_share.R

```diff
@@ -255,11 +215,37 @@ target_market_share <- function(data,
     ) %>%
     select(-.data$target_name, -.data$green_or_brown)
 
+  if (!by_company) {
+    aggregate_company_groups <- c(
+      "sector_ald",
+      "technology",
+      "year",
+      "scenario",
+      "region",
+      "scenario_source"
+    )
+
+    data <- data %>%
+      group_by(!!!rlang::syms(aggregate_company_groups)) %>%
+      summarize(
+        weighted_production = sum(.data$weighted_production),
+        weighted_production_target = sum(.data$weighted_production_target),
+        weighted_technology_share = sum(.data$weighted_technology_share)
+      )
+  }
```

Comment:
this was initially done directly by `summarize_weighted_production`, but by adding `name_ald` to the summary groups, we need to actually summarize again at the end, and potentially aggregate to portfolio level. I think separating the one summary into two distinct summaries, which aim to achieve different things, is better